BlueprintUI/Sources/BlueprintView/BlueprintView.swift

```diff
@@ -179,26 +183,72 @@ public final class BlueprintView: UIView {
             )
         }
 
-        return element.content.measure(
-            in: measurementConstraint(with: size),
+        let constraint = measurementConstraint(with: fittingSize)
+
+        if let cachedSize = sizesThatFit[constraint] {
+            return cachedSize
+        }
+
+        let measurement = element.content.measure(
+            in: constraint,
             environment: makeEnvironment(),
             cache: CacheFactory.makeCache(name: "sizeThatFits:\(type(of: element))")
         )
+
+        sizesThatFit[constraint] = measurement
+
+        return measurement
+    }
+
+    ///
+    /// Measures the size needed to display the view within then given constraining size,
+    /// by measuring the current `element` of the `BlueprintView`.
+    ///
+    /// If you would like to not constrain the measurement in a given axis,
+    /// pass `0.0` or `.greatestFiniteMagnitude` for that axis.
+    ///
+    public override func systemLayoutSizeFitting(
+        _ targetSize: CGSize
+    ) -> CGSize {
+        /// For us, this is the same as `sizeThatFits`, since blueprint does not
+        /// contain the same concept of constraints as Autolayout.
+        sizeThatFits(targetSize)
+    }
+
+    ///
+    /// Measures the size needed to display the view within then given constraining size,
+    /// by measuring the current `element` of the `BlueprintView`.
+    ///
+    /// If you would like to not constrain the measurement in a given axis,
+    /// pass `0.0` or `.greatestFiniteMagnitude` for that axis.
+    ///
+    public override func systemLayoutSizeFitting(
+        _ targetSize: CGSize,
+        withHorizontalFittingPriority horizontalFittingPriority: UILayoutPriority,
+        verticalFittingPriority: UILayoutPriority
+    ) -> CGSize {
+        /// For us, this is the same as `sizeThatFits`, since blueprint does not
+        /// contain the same concept of constraints as Autolayout.
+        sizeThatFits(targetSize)
     }
 
     public override var intrinsicContentSize: CGSize {
-        if let intrinsicContentSize = cachedIntrinsicContentSize {
-            return intrinsicContentSize
+        guard element != nil else {
+            return CGSize(
+                width: UIView.noIntrinsicMetric,
+                height: UIView.noIntrinsicMetric
+            )
         }
 
-        let intrinsicContentSize = resolveIntrinsicContentSize()
-        cachedIntrinsicContentSize = intrinsicContentSize
-        return intrinsicContentSize
-    }
+        let constraint = { () -> SizeConstraint in
+            if bounds.width == 0 {
+                return .unconstrained
+            } else {
+                return .init(width: bounds.width)
+            }
+        }()
```

Comment:
optional nit: IMO immediately invoked closures are hard to read, and this one can be avoided

```suggestion
        let constraint: SizeConstraint = bounds.width.isZero
            ? .unconstrained
            : SizeConstraint(width: bounds.width)
```

Reply:
Will go with an inline `func constraint() -> SizeConstraint {` since Xcode's autocomplete doesn't seem to indent this correctly.